eventstore/mongodb/eventstore.go

```diff
@@ -82,7 +82,7 @@ func NewEventStoreWithSession(session *mgo.Session, database string) (*EventStor
 }
 
 type aggregateRecord struct {
-	AggregateID string         `bson:"_id"`
+	AggregateID interface{}    `bson:"_id"`
```

Comment:
Hmm, maybe unmarshal into bson.Raw as a first step, and then into the concrete ID hidden behind the interface, which in turn have been created by the factory. Much like the event, but with only one ID type (as hinted about in another comment).